src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImpl.java

```diff
@@ -64,21 +58,16 @@ public boolean hasOpenPullRequests(String branchName, BitbucketSCMRepository rep
                 .isPresent();
     }
 
-    /**
-     * retrieves a pull request given ids and keys
-     * @param key
-     * @param slug
-     * @param serverId
-     * @param pullRequestId
-     * @return desired pull request else null
-     */
     @Override
     public Optional<BitbucketPullRequest> getPullRequest(String key, String slug, String serverId, int pullRequestId) {
         PullRequestStoreImpl.CacheKey cacheKey =
                 new PullRequestStoreImpl.CacheKey(key, slug, serverId);
-        for (BitbucketPullRequest b: pullRequests.get(cacheKey)) {
-            if (b.getId() == pullRequestId) {
-                return Optional.of(b);
+        if (!pullRequests.containsKey(cacheKey)) {
```

Comment:
There is a check-then-act race condition here. 
I think what you need to do is essentially
```java
 ConcurrentLinkedQueue<BitbucketPullRequest> prs = pullRequests.get(cacheKey) ;
        if(prs== null || prs.isEmpty()) {
            return  Optional.empty();
        }
        return prs.stream().filter(pr -> pr.getId() == pullRequestId).findFirst();
```